src/azureFunction/azureFunctionUtils.ts

```diff
@@ -78,6 +78,14 @@ export async function setLocalAppSetting(projectFolder: string, key: string, val
 
 export async function getAzureFunctionsExtensionApi(): Promise<AzureFunctionsExtensionApi | undefined> {
 	const apiProvider = await vscode.extensions.getExtension(constants.azureFunctionsExtensionName)?.activate() as AzureExtensionApiProvider;
+	if (!apiProvider) {
+		const response = await vscode.window.showInformationMessage(constants.azureFunctionsExtensionNotFound,
+			constants.installAzureFunction, constants.doNotInstall);
+		if (response === constants.installAzureFunction) {
+			await vscode.env.openExternal(vscode.Uri.parse(constants.linkToAzureFunctionExtension));
```

Comment:
I think it would be better to have 3 buttons

1. `Install` - directly installs extension by calling the `workbench.extensions.installExtension` and passing it the extension ID. I don't believe that a progress dialog is shown while it's installing so it'd be good to wrap that in a vscode.window.withProgress call as well
2. `View Gallery` - opens gallery, ideally to the actual page for the extension. Look at the `workbench.extensions.search` command to see if that'll work, there might be another command you can use too if that doesn't fit our needs
3. `Do not install` - same as what you have